src/components/ListItem.jsx

```diff
@@ -1,5 +1,45 @@
+import { useEffect, useState } from 'react';
+import { updateItem } from '../api/firebase';
+import { ONE_DAY_IN_MILLISECONDS } from '../utils';
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ listPath, item, name }) {
+	const [isChecked, setIsChecked] = useState(false);
+
+	function handleChange(e) {
+		// ensures checked box can be toggled
+		setIsChecked(!isChecked);
+		// pass listPath, copies the item object and sets a new value named isChecked
+		updateItem(listPath, { ...item, isChecked: !isChecked });
+	}
+
+	// this useEffect will run everytime the value of "dateLastPurchased" on the item object changes
+	useEffect(() => {
+		//
+		if (item.dateLastPurchased) {
+			const now = new Date();
+			// calculate expiration time by adding on 24 hours onto the value that is stored on the dateLastPurchased key
+			// this variable will be in milliseconds
+			const expirationTime =
```

Comment:
The reset after 24 hours functionality works correctly!

Reply:
Nice!